### ISSUE TITLE
[BUG] two python tests are failing with numba==0.42.0rc1

### ISSUE THREAD

**Issue by AK-ayush** [rapidsai/cudf#560]
**Description**
Conda env created using `/cudf/conda/environments/dev_py35.yml`, installs the `numba==0.42.0rc1`. And with this numba version, two functional test within `test_dataframe.py` are failing.
* `test_dataframe.test_buffer_append()`
* `test_dataframe.test_series_basic()`  
  
 When i tried downgrading the numba version to 0.40.0 and 0.41.0, all the tests got successfully passed.  

Error log:
```
========================================= FAILURES ==========================================
________________________________________test_buffer_append___________________________________

    def test_buffer_append():
        n = 10
        expected = np.arange(n, dtype=np.float64)
        buf = Buffer(expected, size=n - 4, capacity=n)
        assert buf.size == n - 4
        assert buf.capacity == n
        np.testing.assert_equal(buf.mem.copy_to_host(), expected)
        np.testing.assert_equal(buf.to_array(), np.arange(n - 4, dtype=np.float64))

        # Buffer.append
>       buf.append(1.23)

cudf/tests/test_dataframe.py:43:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cudf/dataframe/buffer.py:155: in append
    self.extend(np.asarray(element, dtype=self.dtype))
cudf/dataframe/buffer.py:161: in extend
    self.mem[self.size:].copy_to_device(array)
/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devices.py:212: in _require_cuda_context
    return fn(*args, **kws)
/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devicearray.py:197: in copy_to_device
    check_array_compatibility(self_core, ary_core)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ary1 = <numba.cuda.cudadrv.devicearray.DeviceNDArray object at 0x7f9e915d94a8>, ary2 = array(1.23)

    def check_array_compatibility(ary1, ary2):
        ary1sq, ary2sq = ary1.squeeze(), ary2.squeeze()
        if ary1.dtype != ary2.dtype:
            raise TypeError('incompatible dtype: %s vs. %s' %
                            (ary1.dtype, ary2.dtype))
        if ary1sq.shape != ary2sq.shape:
            raise ValueError('incompatible shape: %s vs. %s' %
>                            (ary1.shape, ary2.shape))
E           ValueError: incompatible shape: (4,) vs. ()

/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devicearray.py:696: ValueError
________________________________________test_series_basic_______________________________________

    def test_series_basic():
        # Make series from buffer
        a1 = np.arange(10, dtype=np.float64)
        series = Series(a1)
        assert len(series) == 10
        np.testing.assert_equal(series.to_array(), np.hstack([a1]))

        # Add new buffer
        a2 = np.arange(5)
>       series = series.append(a2)

cudf/tests/test_dataframe.py:88:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cudf/dataframe/series.py:436: in append
    return Series(self._column.append(other_col))
cudf/dataframe/column.py:458: in append
    newbuf.extend(buf.to_gpu_array())
cudf/dataframe/buffer.py:161: in extend
    self.mem[self.size:].copy_to_device(array)
/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devices.py:212: in _require_cuda_context
    return fn(*args, **kws)
/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devicearray.py:187: in copy_to_device
    check_array_compatibility(self_core, ary_core)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

ary1 = <numba.cuda.cudadrv.devicearray.DeviceNDArray object at 0x7f9e90c27080>, ary2 = <numba.cuda.cudadrv.devicearray.DeviceNDArray object at 0x7f9e90c27160>

    def check_array_compatibility(ary1, ary2):
        ary1sq, ary2sq = ary1.squeeze(), ary2.squeeze()
        if ary1.dtype != ary2.dtype:
            raise TypeError('incompatible dtype: %s vs. %s' %
                            (ary1.dtype, ary2.dtype))
        if ary1sq.shape != ary2sq.shape:
            raise ValueError('incompatible shape: %s vs. %s' %
>                            (ary1.shape, ary2.shape))
E           ValueError: incompatible shape: (15,) vs. (10,)

/conda/envs/cudf/lib/python3.5/site-packages/numba/cuda/cudadrv/devicearray.py:696: ValueError
```

**Comment by thomcom** [comment]
I actually put some time into fixing this issue in our code - it seems there's a big with numba and shapes in this context, as it was failing when I'd modified the shapes to be (4,) and (1,1), and also (4,) and (1,) if I recall correctly. Fix in the env setup is incoming.

**Comment by thomcom** [comment]
Looks like we're using many features from numba 0.42.0rc1, so this fix is a no-go.

**Comment by kkraus14** [comment]
We use features from 0.40.0 (non-rc) where we should be able to get by with specifying to use only release versions 0.40.0 and newer.

**Comment by thomcom** [comment]
When I updated my numba to 0.40.0 I got lots of failed imports. Maybe something else was broken when I changed the version number? I'll revisit it first thing next morning.

**Comment by kkraus14** [comment]
You should never get failed imports, maybe some failing tests if we're using features from 0.41, but never failed imports unless your environment was corrupted.

**Comment by harrism** [comment]
Any update on this? Trying to decide how to prioritize for 0.5 release.

**Comment by thomcom** [comment]
Sorry, I did this on Friday before New Years and then it was Wednesday. Results are coming up, I'll reopen my PR that sets numba to 0.40.0 if it works out.

**Comment by thomcom** [comment]
I reopened the fix and can confirm the original issue.

**Comment by harrism** [comment]
Fixed by #648.
